ontotext-yasgui-web-component/src/services/yasqe/yasqe-service.ts

```diff
@@ -15,15 +16,57 @@ export class YasqeService {
   //@ts-ignore
   buttonBuilders: Map<string, ((yasguiConfiguration: YasguiConfiguration, yasqe: Yasqe) => HTMLElement | HTMLElement[])> = new Map<string, (() => HTMLElement | HTMLElement[])>();
 
+  private static pluginButtonNameToClassNameMapping: Map<YasqeButtonType, string>;
+
+
   constructor(serviceFactory: ServiceFactory) {
     this.eventService = serviceFactory.getEventService();
+    this.initPluginButtonNameToClassNameMapping();
     this.translationService = serviceFactory.get(TranslationService);
-    this.buttonBuilders.set('createSavedQuery', () => this.buildCreateSaveQueryButton());
-    this.buttonBuilders.set('showSavedQueries', () => this.buildShowSavedQueriesButton());
-    this.buttonBuilders.set('shareQuery', () => this.buildShareQueryButton());
+    this.buttonBuilders.set(YasqeButtonName.CREATE_SAVED_QUERY, () => this.buildCreateSaveQueryButton());
+    this.buttonBuilders.set(YasqeButtonName.SHOW_SAVED_QUERIES, () => this.buildShowSavedQueriesButton());
+    this.buttonBuilders.set(YasqeButtonName.SHARE_QUERY, () => this.buildShareQueryButton());
     this.buttonBuilders.set('includeInferredStatements', (externalConfiguration, yasqe) => this.buildInferAndSameAsButtons(externalConfiguration, yasqe));
   }
 
+  private initPluginButtonNameToClassNameMapping() {
+    YasqeService.pluginButtonNameToClassNameMapping = new Map();
+    YasqeService.pluginButtonNameToClassNameMapping.set(YasqeButtonName.CREATE_SAVED_QUERY, `yasqe_${YasqeButtonName.CREATE_SAVED_QUERY}Button`);
+    YasqeService.pluginButtonNameToClassNameMapping.set(YasqeButtonName.SHOW_SAVED_QUERIES, `yasqe_${YasqeButtonName.SHOW_SAVED_QUERIES}Button`);
+    YasqeService.pluginButtonNameToClassNameMapping.set(YasqeButtonName.SHARE_QUERY, `yasqe_${YasqeButtonName.SHARE_QUERY}Button`);
+    YasqeService.pluginButtonNameToClassNameMapping.set(YasqeButtonName.EXPANDS_RESULTS, `yasqe_${YasqeButtonName.EXPANDS_RESULTS}Button`);
+    YasqeService.pluginButtonNameToClassNameMapping.set(YasqeButtonName.INFER_STATEMENTS, `yasqe_${YasqeButtonName.INFER_STATEMENTS}Button`);
+
+  }
+
+  static getActionButtonClassName(actionButtonName: YasqeButtonType) {
+    return this.pluginButtonNameToClassNameMapping.get(actionButtonName);
```

Comment:
Maybe check if it exists and do some logging if it doesn't

Reply:
Done